src/main/fc/settings.c

```diff
@@ -719,6 +725,12 @@ const clivalue_t valueTable[] = {
     { "esc_sensor_halfduplex",          VAR_UINT8  | MASTER_VALUE | MODE_LOOKUP, .config.lookup = { TABLE_OFF_ON }, PG_ESC_SENSOR_CONFIG, offsetof(escSensorConfig_t, halfDuplex) },
 #endif
     { "led_inversion",                  VAR_UINT8  | MASTER_VALUE, .config.minmax = { 0, ((1 << STATUS_LED_NUMBER) - 1) }, PG_STATUS_LED_CONFIG, offsetof(statusLedConfig_t, inversion) },
+#ifdef MAG
+#ifdef USE_MAG_HMC5883
+    { "mag_hmc5883_bustype",            VAR_UINT8  | MASTER_VALUE | MODE_LOOKUP, .config.lookup = { TABLE_BUS_TYPE }, PG_BUSDEV_HMC5883_CONFIG, offsetof(busDeviceConfig_t, busType) },
+    { "mag_hmc5883_busnum",             VAR_UINT8  | MASTER_VALUE, .config.minmax = { 0, I2CDEV_COUNT - 1 }, PG_BUSDEV_HMC5883_CONFIG, offsetof(busDeviceConfig_t, busNum) },
```

Comment:
@ledvinap , yes agreed `spiWriteCommand` should eventually be merged with I2C write. But I think we should defer that until after the SPI changes have been made. Having too much going on at one time can lead to silly errors.